### ISSUE TITLE
why add_definitions("-std=c++11")

### ISSUE THREAD

**Issue by LeeRuns** [conan-io/cmake-conan#396]
After looking at this, I can't help but think the point is to trigger the CPP 11 std lib. But that makes no sense, so why is this part of the example?

**Comment by czoido** [comment]
Hi @LeeRuns,
The `add_definitions("-std=c++11")` is there because the required library needs to set the language standard for some compilers (those that have a standard lower than that) because some features of that standard are used by the **fmt** library that is required by the example. 
Anyway I think that it would be better to change that to `set_property(TARGET main PROPERTY CXX_STANDARD 11)`

**Comment by LeeRuns** [comment]
If the point is flexibility then shouldn't we always just get this from the cmake file? By placing it inside of the conan section of my cmake file it makes it appear to the reader that the standard must be set to cpp11 which is not true. I really think this should be removed. 


As an aside...
I am having the hardest time straightening out how to link std::__cxx linker and I know it must be related to the setup between cmake and conan. I am running windows, and using gnu/gcc via msys. I get it working occasionally and then poof, linker error appears that is totally unclear. I know this is related to the ABI, any chance you have experience with solving this @czoido

**Comment by czoido** [comment]
Hi,

> If the point is flexibility then shouldn't we always just get this from the cmake file? By placing it inside of the conan section of my cmake file it makes it appear to the reader that the standard must be set to cpp11 which is not true. I really think this should be removed.
> 

Do you mean to remove this? https://github.com/conan-io/cmake-conan/blob/1ed947da9c0207b070c7967af1c60da636039c33/CMakeLists.txt#L7 
If you do that this example will not build for certain compilers that don't use a lower standard than 11 as the default.

> As an aside... I am having the hardest time straightening out how to link std::__cxx linker and I know it must be related to the setup between cmake and conan. I am running windows, and using gnu/gcc via msys. I get it working occasionally and then poof, linker error appears that is totally unclear. I know this is related to the ABI, any chance you have experience with solving this @czoido

Can you provide a minimal reproducible example for this?

**Comment by LeeRuns** [comment]
I was able to fix this. I can only explain that it has to do with 3 independent entities working separately:

1. conan io
2. gcc
3. cpp std committee
4. cmake


As a result, to get a clear setup I had to define the lisbstdcpp to be libstdc++11 in each of my compiler commands and ensure that this went through the whole project then change the same for conan. This is the new standard, don't understand why conan is wanting to use old abi. But read other issues, and seems like that is a topic for another place.

Hopefully that helps others who go through this process.
